actions/apps.ts

```diff
@@ -40,11 +40,10 @@ export function doAppCall<Res=unknown>(call: AppCall): ActionFunc {
                 return {data: res};
             }
 
-            if (call.context.location === AppsBindings.COMMAND && call.type === AppCallTypes.FORM) {
-                return {data: res};
+            if (call.type === AppCallTypes.SUBMIT) {
```

Comment:
I think I understand the logic here, but does not feel intuitive. Correct me if I am wrong:
- If the response is a form is because I have:
  - Submitted a call
  - Made a lookup
- If it is a call submission, I have to open the modal. In other cases the modal will just be updated

Am I right? Is there any way to make this more intuitive?

Reply:
@larkox 
The response type is `form` if:

- We are fetching a form for a command. The user has typed a full sub command, and then we fetch the form for the subcommand via `call.type = 'form'`. The call `response.type = 'form'`
- We submit a form, and the app wants a modal to open as a result. So `call.type = 'submit' && response.type = 'form'` means open a new modal
- We select a significant value in a form (one that has `refresh: true` in the field's definition). Similar to fetching a command's form, the `call.type = 'form'` and the `response.type = 'form'`. `call.values.name` is also present, signaling that a value has been selected. I'm thinking we should have a `call.type = 'refresh'` for this case instead of `form`.

Lookups are different in that they just return an array of values like `{label: 'Label1', value: 'Value1'}`. In this case `call.type = 'lookup'` and `response.type = 'ok'`